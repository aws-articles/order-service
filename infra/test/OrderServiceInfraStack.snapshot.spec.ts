import {SynthUtils} from "@aws-cdk/assert";
import {OrderServiceInfraStack} from "../lib/OrderServiceInfraStack";
import * as cdk from "@aws-cdk/core";

test("should match the order service infra snapshot", () => {
    const app = new cdk.App();
    const stack = new OrderServiceInfraStack(app, "OrderServiceStack");
    const cloudFormationStack = SynthUtils.synthesize(stack);

    expect(cloudFormationStack).toMatchSnapshot();
});