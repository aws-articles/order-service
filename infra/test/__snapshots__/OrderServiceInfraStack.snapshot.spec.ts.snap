// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the order service infra snapshot 1`] = `
CloudFormationStackArtifact {
  "_dependencyIDs": Array [],
  "_deps": Array [],
  "assembly": CloudAssembly {
    "artifacts": Array [
      [Circular],
      TreeCloudArtifact {
        "_dependencyIDs": Array [],
        "_deps": Array [],
        "assembly": [Circular],
        "file": "tree.json",
        "id": "Tree",
        "manifest": Object {
          "properties": Object {
            "file": "tree.json",
          },
          "type": "cdk:tree",
        },
        "messages": Array [],
      },
    ],
    "directory": "/var/folders/v_/nv8xrk59535_k3p2sl_k288m0000gn/T/cdk.outRCaMyP",
    "manifest": Object {
      "artifacts": Object {
        "OrderServiceStack": Object {
          "environment": "aws://unknown-account/unknown-region",
          "metadata": Object {
            "/OrderServiceStack": Array [
              Object {
                "data": Object {
                  "artifactHashParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C",
                  "id": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
                  "packaging": "zip",
                  "path": "asset.b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
                  "s3BucketParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
                  "s3KeyParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
                  "sourceHash": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
                },
                "trace": Array [
                  "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:533:17)",
                  "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
                  "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:asset",
              },
            ],
            "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/ArtifactHash": Array [
              Object {
                "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C",
                "trace": Array [
                  "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:26:36)",
                  "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
                  "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
                  "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:logicalId",
              },
            ],
            "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/S3Bucket": Array [
              Object {
                "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
                "trace": Array [
                  "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:16:32)",
                  "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
                  "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
                  "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:logicalId",
              },
            ],
            "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/S3VersionKey": Array [
              Object {
                "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
                "trace": Array [
                  "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:21:31)",
                  "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
                  "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
                  "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:logicalId",
              },
            ],
            "/OrderServiceStack/order-service-lambda/Resource": Array [
              Object {
                "data": "orderservicelambdaA48E3679",
                "trace": Array [
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:445:35)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:logicalId",
              },
            ],
            "/OrderServiceStack/order-service-lambda/ServiceRole/Resource": Array [
              Object {
                "data": "orderservicelambdaServiceRoleC37326C6",
                "trace": Array [
                  "new Role (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-iam/lib/role.ts:300:18)",
                  "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:432:31)",
                  "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
                  "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
                  "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
                  "new Promise (<anonymous>)",
                  "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
                  "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
                  "processTicksAndRejections (internal/process/task_queues.js:93:5)",
                ],
                "type": "aws:cdk:logicalId",
              },
            ],
          },
          "properties": Object {
            "templateFile": "OrderServiceStack.template.json",
          },
          "type": "aws:cloudformation:stack",
        },
        "Tree": Object {
          "properties": Object {
            "file": "tree.json",
          },
          "type": "cdk:tree",
        },
      },
      "runtime": Object {
        "libraries": Object {
          "jsii-runtime": "node.js/v13.2.0",
        },
      },
      "version": "1.16.0",
    },
    "runtime": Object {
      "libraries": Object {
        "jsii-runtime": "node.js/v13.2.0",
      },
    },
    "version": "1.16.0",
  },
  "assets": Array [
    Object {
      "artifactHashParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C",
      "id": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
      "packaging": "zip",
      "path": "asset.b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
      "s3BucketParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
      "s3KeyParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
      "sourceHash": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
    },
  ],
  "displayName": "OrderServiceStack",
  "environment": Object {
    "account": "unknown-account",
    "name": "aws://unknown-account/unknown-region",
    "region": "unknown-region",
  },
  "id": "OrderServiceStack",
  "manifest": Object {
    "environment": "aws://unknown-account/unknown-region",
    "metadata": Object {
      "/OrderServiceStack": Array [
        Object {
          "data": Object {
            "artifactHashParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C",
            "id": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
            "packaging": "zip",
            "path": "asset.b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
            "s3BucketParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
            "s3KeyParameter": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
            "sourceHash": "b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60",
          },
          "trace": Array [
            "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:533:17)",
            "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
            "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:asset",
        },
      ],
      "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/ArtifactHash": Array [
        Object {
          "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C",
          "trace": Array [
            "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:26:36)",
            "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
            "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
            "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:logicalId",
        },
      ],
      "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/S3Bucket": Array [
        Object {
          "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
          "trace": Array [
            "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:16:32)",
            "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
            "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
            "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:logicalId",
        },
      ],
      "/OrderServiceStack/AssetParameters/b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60/S3VersionKey": Array [
        Object {
          "data": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
          "trace": Array [
            "new FileAssetParameters (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/private/asset-parameters.ts:21:31)",
            "OrderServiceInfraStack.addFileAsset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/core/lib/stack.ts:520:16)",
            "new Asset (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-s3-assets/lib/asset.ts:113:28)",
            "AssetCode.bind (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/code.ts:180:20)",
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:442:29)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:logicalId",
        },
      ],
      "/OrderServiceStack/order-service-lambda/Resource": Array [
        Object {
          "data": "orderservicelambdaA48E3679",
          "trace": Array [
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:445:35)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:logicalId",
        },
      ],
      "/OrderServiceStack/order-service-lambda/ServiceRole/Resource": Array [
        Object {
          "data": "orderservicelambdaServiceRoleC37326C6",
          "trace": Array [
            "new Role (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-iam/lib/role.ts:300:18)",
            "new Function (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/@aws-cdk/aws-lambda/lib/function.ts:432:31)",
            "new OrderServiceInfraStack (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/lib/OrderServiceInfraStack.ts:14:9)",
            "Object.<anonymous> (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/test/OrderServiceInfraStack.snapshot.spec.ts:7:19)",
            "Object.asyncJestTest (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:102:37)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:43:12",
            "new Promise (<anonymous>)",
            "mapper (/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:26:19)",
            "/Users/sarthakmakhija/Projects/aws-articles/order-service/infra/node_modules/jest-jasmine2/build/queueRunner.js:73:41",
            "processTicksAndRejections (internal/process/task_queues.js:93:5)",
          ],
          "type": "aws:cdk:logicalId",
        },
      ],
    },
    "properties": Object {
      "templateFile": "OrderServiceStack.template.json",
    },
    "type": "aws:cloudformation:stack",
  },
  "messages": Array [],
  "name": "OrderServiceStack",
  "originalName": "OrderServiceStack",
  "parameters": Object {},
  "stackName": "OrderServiceStack",
  "template": Object {
    "Parameters": Object {
      "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60ArtifactHash133A0B3C": Object {
        "Description": "Artifact hash for asset \\"b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60\\"",
        "Type": "String",
      },
      "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9": Object {
        "Description": "S3 bucket for asset \\"b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60\\"",
        "Type": "String",
      },
      "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3": Object {
        "Description": "S3 key for asset version \\"b0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60\\"",
        "Type": "String",
      },
    },
    "Resources": Object {
      "orderservicelambdaA48E3679": Object {
        "DependsOn": Array [
          "orderservicelambdaServiceRoleC37326C6",
        ],
        "Properties": Object {
          "Code": Object {
            "S3Bucket": Object {
              "Ref": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3BucketE36DCFE9",
            },
            "S3Key": Object {
              "Fn::Join": Array [
                "",
                Array [
                  Object {
                    "Fn::Select": Array [
                      0,
                      Object {
                        "Fn::Split": Array [
                          "||",
                          Object {
                            "Ref": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
                          },
                        ],
                      },
                    ],
                  },
                  Object {
                    "Fn::Select": Array [
                      1,
                      Object {
                        "Fn::Split": Array [
                          "||",
                          Object {
                            "Ref": "AssetParametersb0612dabf009a9a682fd83373a4d19e9bf32384f7222b4018ae420ea6d3f4e60S3VersionKey819BE4F3",
                          },
                        ],
                      },
                    ],
                  },
                ],
              ],
            },
          },
          "FunctionName": "order-service-lambda",
          "Handler": "handler.ordersHandler",
          "Role": Object {
            "Fn::GetAtt": Array [
              "orderservicelambdaServiceRoleC37326C6",
              "Arn",
            ],
          },
          "Runtime": "nodejs10.x",
        },
        "Type": "AWS::Lambda::Function",
      },
      "orderservicelambdaServiceRoleC37326C6": Object {
        "Properties": Object {
          "AssumeRolePolicyDocument": Object {
            "Statement": Array [
              Object {
                "Action": "sts:AssumeRole",
                "Effect": "Allow",
                "Principal": Object {
                  "Service": "lambda.amazonaws.com",
                },
              },
            ],
            "Version": "2012-10-17",
          },
          "ManagedPolicyArns": Array [
            Object {
              "Fn::Join": Array [
                "",
                Array [
                  "arn:",
                  Object {
                    "Ref": "AWS::Partition",
                  },
                  ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
                ],
              ],
            },
          ],
        },
        "Type": "AWS::IAM::Role",
      },
    },
  },
  "templateFile": "OrderServiceStack.template.json",
}
`;
